src/utils/getAvatarUrl.ts

```diff
@@ -4,12 +4,47 @@
  */
 
 import { generateUrl } from '@nextcloud/router'
+import { checkIfDarkTheme } from '../functions/isDarkTheme/index.ts'
 
-export const getAvatarUrl = (user: string, size: number | string, isGuest?: boolean): string => {
-	const darkTheme = window.getComputedStyle(document.body)
-		.getPropertyValue('--background-invert-if-dark') === 'invert(100%)'
+interface AvatarUrlOptions {
+	/**
+	 * Should the dark theme variant be used.
+	 */
+	isDarkTheme?: boolean
 
-	return generateUrl('/avatar' + (isGuest ? '/guest' : '') + '/{user}/{size}' + (darkTheme ? '/dark' : ''), {
+	/**
+	 * Is the user a guest user.
+	 */
+	isGuest?: boolean
+
+	/**
+	 * Size of the avatar.
+	 * @default 64
+	 */
+	size?: number
```

Comment:
```suggestion
	size?: 64 | 512
```

but then your .spec.ts are going to complain, right?)